file-formats/aifi/type/zif_aff_aifi_v1.intf.abap

```diff
@@ -0,0 +1,506 @@
+      "! $required
+      " bgPF (type /aif/bgp_opid_e)
+      operation_id         TYPE zif_aff_types_v1=>ty_object_name_30,
+      "! <p class="shorttext">Operation Variant ID</p>
+      "! Operation variant id
+      "! $required
+      " bgPF (type /aif/bgp_variant_id_e)
+      operation_variant_id TYPE n LENGTH 10,
+      "! <p class="shorttext">Operation Variant</p>
+      "! Operation variant
+      "! $required
+      " bgPF (type /aif/bgp_variant_e)
+      operation_varaint    TYPE string,
+    END OF ty_bgpf_settings.
+
+  TYPES:
+    "! <p class="shorttext">Application Interface</p>
+    "! Application interface
+    BEGIN OF ty_aif_interface,
+      "! <p class="shorttext">Integration Type</p>
+      "! Integration type
+      integration_type         TYPE ty_integration_type,
+      "! <p class="shorttext">SOAP Settings</p>
+      "! SOAP settings
+      soap_settings            TYPE ty_soap_settings,
+      "! <p class="shorttext">Event Settings</p>
+      "! Event settings
+      event_settings           TYPE ty_event_settings,
+      "! <p class="shorttext">BgPF Settings</p>
+      "! BgPF settings
+      bgpf_settings            TYPE ty_bgpf_settings,
+      "! <p class="shorttext">Namespace</p>
+      "! Namespace
+      "! $required
+      " (type /aif/ns_obj_name)
+      namespace                TYPE c LENGTH 15,
+      "! <p class="shorttext">AIF Interface Name</p>
+      "! AIF interface name
+      "! $required
+      " (type /aif/ifname)
+      interface_name           TYPE c LENGTH 10,
+      "! <p class="shorttext">AIF Interface Version</p>
+      "! AIF interface version
+      "! $required
+      " (type /aif/ifversion)
+      interface_version        TYPE c LENGTH 5,
+      "! <p class="shorttext">RAW Structure</p>
+      "! RAW structure
+      " (type /aif/if_ddic_struct_raw)
+      raw_structure            TYPE zif_aff_types_v1=>ty_object_name_30,
+      "! <p class="shorttext">SAP Structure</p>
+      "! SAP structure
+      " (type /aif/if_ddic_struct)
+      sap_structure            TYPE zif_aff_types_v1=>ty_object_name_30,
+      "! <p class="shorttext">Index Table</p>
+      "! Index table
+      "! $default '/AIF/STD_IDX_TBL'
+      " (type /aif/msg_tbl)
+      index_table              TYPE zif_aff_types_v1=>ty_object_name_30,
+      "! <p class="shorttext">Interface Display Name</p>
+      "! Interface display name
+      " (type /aif/ifdisplay)
+      display_name             TYPE c LENGTH 120,
+      "! <p class="shorttext">Application Log Lifetime</p>
+      "! Application log lifetime
+      "! $default '90'
+      " (type /aif/msg_lifetime)
+      app_log_lifetime         TYPE n LENGTH 10,
+      "! <p class="shorttext">Direction</p>
+      "! Direction
+      " (type /aif/ifdirection)
+      direction                TYPE ty_direction_type,
+      "! <p class="shorttext">Preprocessing</p>
+      "! Preprocessing
+      " /AIF/PRE_PROCESSING
+      preprocessing            TYPE abap_bool,
```

Comment:
A suggestion from our side: For booleans we usually prefer the prefix `is_` to indicate a boolean in the json. And maybe for the `proxy_xml_transformation` a `uses_` prefix? 

But this is up to you.

Reply:
I prefer to follow best practice and common format, so I will adjust it.